cloudwatch_logs_common/test/log_batch_test.cpp

```diff
@@ -0,0 +1,246 @@
+
+  std::shared_ptr<TestPublisher> test_publisher;
+  std::shared_ptr<LogBatcher> batcher;
+  std::shared_ptr<LogService> cw_service;
+
+  std::shared_ptr<TaskObservedQueue<LogCollection>> stream_data_queue;
+  std::shared_ptr<Aws::DataFlow::QueueMonitor<TaskPtr<LogCollection>>>queue_monitor;
+};
+
+auto log_comparison = [](const LogType & log1, const LogType & log2)
+  { return log1.GetTimestamp() < log2.GetTimestamp(); };
+
+/**
+ * Test File Manager
+ */
+class TestLogFileManager : public FileManager<LogCollection>, public Waiter
+{
+public:
+
+    TestLogFileManager() : FileManager(nullptr) {
+      written_count.store(0);
+    }
+
+    void write(const LogCollection & data) override {
+      last_data_size = data.size();
+      written_count++;
+      this->notify();
+    };
+
+    //test that the readBatch function works with 24 hour interval
+    FileObject<LogCollection> readBatch(size_t batch_size) override {
+      std::priority_queue<std::tuple<long, std::string, uint64_t>> pq;
+      long latestTime = 0;
+      for (size_t i = 0; i < batch_size; ++i) {
+        std::string line;
+        if(i == batch_size-1){
+            line = "{\"timestamp\":" + std::to_string(i + 86400000) + ",\"message\":\"Last log in batch file\"}";
+        }
+        else{
+            line = "{\"timestamp\":" + std::to_string(i) + ",\"message\":\"Testing batch file\"}";
+        }
+        Aws::String aws_line(line.c_str());
+        Aws::Utils::Json::JsonValue value(aws_line);
+        Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
+        pq.push(std::make_tuple(input_event.GetTimestamp(), line, 0));
+        if(input_event.GetTimestamp() > latestTime){
+          latestTime = input_event.GetTimestamp();
+        }
+      }
+
+      std::set<LogType, decltype(log_comparison)> log_set(log_comparison);
+      size_t actual_batch_size = 0;
+      while(!pq.empty()){
+        long curTime = std::get<0>(pq.top());
+        std::string line = std::get<1>(pq.top());
+        if(latestTime - curTime < 86400000){
+          Aws::String aws_line(line.c_str());
+          Aws::Utils::Json::JsonValue value(aws_line);
+          Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
+          actual_batch_size++;
+          log_set.insert(input_event);
+        }
+        pq.pop();
+      }
+
+      LogCollection log_data(log_set.begin(), log_set.end());
+      FileObject<LogCollection> file_object;
+      file_object.batch_data = log_data;
+      file_object.batch_size = actual_batch_size;
+      return file_object;
+    }
+
+    std::atomic<int> written_count{};
+    std::atomic<size_t> last_data_size{};
+    std::condition_variable cv;
+    mutable std::mutex mtx;
+};
+
+TEST_F(LogBatchTest, Sanity) {
+  ASSERT_TRUE(true);
+}
+
+TEST_F(LogBatchTest, batch_test_24hours) {
+  std::shared_ptr<TestLogFileManager> fileManager = std::make_shared<TestLogFileManager>();
```

Comment:
If you're instantiating the `TestLogFIleManager` here, I don't understand why all of the pipeline testing components are needed.